### ISSUE TITLE
Cursor changes to arrow when switching windows (macOS)

### ISSUE THREAD

**Issue by jhuckaby** [microsoft/vscode#97864]
## Summary:

When I switch between VSCode windows, the macOS cursor gets stuck on the "Arrow", and does not return to the proper "I-Beam" cursor.  This behavior started in VSCode v1.45, but is also reproducible in v1.45.1.

## Steps:

1. Activate VSCode application.  Notice that the cursor is the proper "I-Beam" cursor.
2. Switch to another VSCode window either using the "Window" menu, or a keyboard shortcut such as `Cmd-Tilde`.
3. The cursor will be an "Arrow" cursor on the new window, and will remain that way.  It gets "stuck" in this state.

## Screencast illustrating the issue:

https://youtu.be/XLCP7F811Zs

## System Information

- macOS Catalina 10.15.3 (19D76)
- VSCode v1.45 & v1.45.1

CPUs | Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)
-- | --
GPU Status | 2d_canvas: enabledflash_3d: enabledflash_stage3d: enabledflash_stage3d_baseline: enabledgpu_compositing: enabledmetal: disabled_offmultiple_raster_threads: enabled_onoop_rasterization: disabled_offprotected_video_decode: unavailable_offrasterization: enabledskia_renderer: disabled_off_okvideo_decode: enabledviz_display_compositor: enabled_onviz_hit_test_surface_layer: disabled_off_okwebgl: enabledwebgl2: enabled
Load (avg) | 2, 2, 2
Memory (System) | 32.00GB (12.30GB free)
Process Argv | -psn_0_8435723
Screen Reader | no
VM | 0%

## Enabled Extensions

Extension | Author (truncated) | Version
-- | -- | --
project-manager | ale | 11.0.0
spellright | ban | 3.0.50
php-symbols | lin | 2.1.0
sublime-keybindings | ms- | 4.0.7
vscode-icons | vsc | 10.1.1
vscode-todo-highlight | way | 1.0.4
material-theme | zhu | 3.6.1

**Comment by alexdima** [comment]
This happens since a very long time for me, the easiest repro is to right click, which turns the cursor to an arrow, and then a mouse leave + enter over the editor is needed to get to be an `I` again.

**Comment by jhuckaby** [comment]
Just confirming this is still reproducible in VSCode 1.49.1.

**Comment by bsides** [comment]
I can confirm it still happens in macOS 12.2, vscode 1.64.2 (Universal). It's quite annoying. Is there anything we could do to make this go away?

**Comment by stdrew** [comment]
Fix coming?  Thanks.

**Comment by alexrs84** [comment]
The only way to recover the `I` beam cursor icon is to trigger a `mouseenter` in the editor again (line number column, or a tab header is far enough to leave the editor). 

![ibeam](https://user-images.githubusercontent.com/1809932/187060445-2af6b1e9-2991-41e7-b704-ea6d91f309c2.gif)


VS Code version: Code 1.70.2 (e4503b30fc78200f846c62cf8091b76ff5547662, 2022-08-16T05:36:37.829Z)
OS version: Darwin x64 21.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.98GB free)|
|Process Argv|--crash-reporter-id 6048ceb8-ffc5-4877-8f7c-cee23ec9bc4a|
|Screen Reader|no|
|VM|0%|
</details>
Extensions: none

**Comment by sebastiansson** [comment]
How can this still be an issue? Still present on the latest version as of today:

Version: 1.76.0 (Universal)
Commit: 92da9481c0904c6adfe372c12da3b7748d74bdcb
Date: 2023-03-01T10:23:45.993Z (1 day ago)
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 21.2.0
Sandboxed: No

**Comment by vhladiienko** [comment]
> How can this still be an issue? Still present on the latest version as of today:

This is likely not a VS Code issue. Happens to me randomly in macOS with different apps.

**Comment by sebastiansson** [comment]
> > How can this still be an issue? Still present on the latest version as of today:
> 
> This is likely not a VS Code issue. Happens to me randomly in macOS with different apps.

Yes it is, the apps you are refering to is probably also Electron apps.

**Comment by vhladiienko** [comment]
> Yes it is, the apps you are referring to is probably also Electron apps.

Finder is not an Electron app. Even if it was, according to your logic it would likely mean this is an Electron or Chromium issue, not a VS Code problem.

**Comment by sebastiansson** [comment]
> Finder is not an Ele

Finder has nothing to do with this issue.

**Comment by sebastiansson** [comment]
> > Yes it is, the apps you are referring to is probably also Electron apps.
> 
> Finder is not an Electron app. Even if it was, according to your logic it would likely mean this is an Electron or Chromium issue, not a VS Code problem. Just Google this problem, it's all over Reddit and Apple's support.

Finder has nothing to do with this issue.

**Comment by vhladiienko** [comment]
> > > Yes it is, the apps you are referring to is probably also Electron apps.
> > 
> > 
> > Finder is not an Electron app. Even if it was, according to your logic it would likely mean this is an Electron or Chromium issue, not a VS Code problem. Just Google this problem, it's all over Reddit and Apple's support.
> 
> Finder has nothing to do with this issue.

Of course, it doesn't. Please re-read your own comment and my reply again.

**Comment by LukePeters** [comment]
I've seen this issue for at least a few years on several versions of macOS and VSCode. It's quite distracting.

**Comment by ac50934** [comment]
Randomly started getting this issue too on MacOs 14.2.1, the real problem being I cannot get the hover previews to display properly (a very functional impact)

Version: 1.86.2 (Universal)
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:42:13.651Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.2.0

**Comment by lorisantiagocm** [comment]
i'm having a similar issue. Ubuntu 22.04.3

**Comment by CedricBorreguin** [comment]
I recently ran into this bug myself. Not a solution but after trying a bunch of things: restarting my computer did it. 

It had more than a week of running continuously, not sure if that could break something within Electron because even restarting VS Code didn't do anything until I restarted the whole computer.

I hate that my suggestion is basically "Have you tried turning it off?", but it really worked.

System info:
MacBook Pro w/ M3 Pro
macOS Sonoma 14.8.5 (23J423)

Version: 1.120.0 (Universal)
Commit:[0958016]( https://github.com/microsoft/vscode/commit/0958016b2af9f09bb4257e0df4a95e2f90590f9f)
Date: 2026-05-12T20:17:22Z
Electron: 39.8.8
ElectronBuildId: 13870025
Chromium: 142.0.7444.265
Node.js: 22.22.1
V8: 14.2.231.22-electron.0
OS: Darwin arm64 23.6.0
